frontend/catalyst/jax_primitives.py

```diff
@@ -368,6 +403,133 @@ def _print_lowering(jax_ctx: mlir.LoweringRuleContext, *args, string=None, memre
     return PrintOp(val=val, const_val=None, print_descriptor=memref).results
 
 
+#
+# transform dialect lowering
+#
+
+# Note: we currently only support peephole passes on function ops
+
+transform_func_type = ir.OpaqueType.get("transform", 'op<"func.func">', ir.Context())
+
+
+#
+# transform_named_sequence
+#
+@transform_named_sequence_p.def_abstract_eval
+def _transform_named_sequence_p_abstract_eval(*args):
+    return ()
+
+
+@transform_named_sequence_p.def_impl
+def _transform_named_sequence_p_def_impl(*args):  # pragma: no cover
+    raise NotImplementedError()
+
+
+def _transform_named_sequence_lowering(jax_ctx: mlir.LoweringRuleContext, *args):
+    module = jax_ctx.module_context.module
+
+    # If there already is a transform.named_sequence in the module, don't add another one!
+    # Do nothing and exit!
+    for i in range(len(module.body.operations)):
+        if module.body.operations[i].operation.name == "transform.named_sequence":
+            return module.body.operations[i].operation.results
+
+    functype = ir.FunctionType.get(inputs=[transform_func_type], results=[])
```

Comment:
Do you know what the input is?

Reply:
We restrict all the passes to be function passes.